New branch28/4

Nowhere4Nothing opened the pull request:
Updated previous points from Stuart

-added birth date as a datetime
- changed names to be more appropriate
- changed optimization in extract patient info based on patient name (need to do more here, I'm not happy with it)
- changed  validate_dicom to not be overly restrictive 

hopefully this is a bit better

## Summary by Sourcery

Refactor DICOM file reading, data extraction, validation, and image handling into dedicated utility modules (`dicom_utils.py`, `inputs_and_outputs.py`). Update the UI to utilize these modules, improving code structure and error handling.

Enhancements:
- Centralize patient data extraction using a `PatientInfo` dataclass.
- Add validation for required DICOM fields and specific image types.
- Parse patient birth date into a standard datetime format.
- Improve normalization of pixel data for image display.
- Improve error handling during file reading and image display in the UI.
- Replace direct DICOM attribute access in the UI with calls to new utility functions.
- Extract image conversion logic from the UI into utility functions (`numpy_to_qimage`, `get_qimage_from_dicom_file`).

Chores:
- Integrate logging for better error tracking in file operations.

Nowhere4Nothing committed:
fixing the fuck up of IDE settings

Redon it without the IDE settings

Nowhere4Nothing committed:
Merge branch 'UOW-Group17:dev' into test

Nowhere4Nothing committed:
little changes forgot new branch

Nowhere4Nothing committed:
Merge branch 'UOW-Group17:dev' into test

Nowhere4Nothing committed:
Merge branch 'UOW-Group17:dev' into test

Nowhere4Nothing committed:
merging and fix

Nowhere4Nothing committed:
Updating from previous pull request

Updated previous points from Stuart

-added birth date as a datetime
- changed names to be more appropriate
- changed optimization in extract patient info (need to do more here, I'm not happy with it)
- changed  validate_dicom to not be overly restrictive

sjswerdloff commented on src/__pycache__/dicom_utils.cpython-311.pyc:
still need to remove this file

sjswerdloff commented on src/dicom_utils.py:
```diff
@@ -0,0 +1,135 @@
+    sex = "Anonymous"
+
+    # If statement to return data - Correct the condition here
+    if not patient_name or len(str(patient_name)) <= 20:
```
If you were to use my name in DICOM format, it would be Swerdloff^Stuart^Julian
(which is 9+1+6+1+6 =23 characters long).
If the name was anonymised, odds are the length will either the length of a UUID (old style anonymisation) or closer to 64 characters (possibly 65, which is a bug because PN is apparently supposed to be limited to 64, even if there are multiple representations/alphabetic/ideographic etc).

sjswerdloff replied:
A pseudonymised name would fit the structure, that's one of the needs (so that a pseudonymised patient data set can be stored in a PACS or Vendor Neutral Archive (VNA)).  So it's a bit challenging.  On the other hand, if the patient name has been pseudonymised it might as well get displayed with the values present.  I don't think there is a particular requirement that a pseudonymised or anonymised name *not* be shown.  But it is a useful thing to show "Anonymous" rather than nothing if there is no PatientName value in the DICOM data.  So perhaps checking "the structure" would be better.

sjswerdloff commented on src/mini_project_ui.py:
```diff
@@ -138,42 +127,50 @@ def set_label(self, ds, field, label, default):
-        
+        if metadata.birth_date:
+            # Change the date variable to a string
+            formatted_birth_date = metadata.birth_date.strftime("%Y-%m-%d")
```
I had real problems in airports in the USA when my birthdate was entered NZ/AU style in to a USA database, so my month and day of birth got swapped (both numbers are valid as month or day).
So... Either the order of numbers has to be displayed to the user (i.e. text showing that it is YYYY-MM-DD) or it should be localised (USA style is MMM-DD-YYYY, i.e. the month three letters is used rather than the numeric value).
This is one option for localisation:
http://stackoverflow.com/questions/985505/ddg#32785195

and it seems to be the best.

sjswerdloff replied:
It's a good thing I had a good relationship with my travel agent.
She was appropriately apologetic and we just moved on.
If a young child is involved, swapping month and day could be *really* bad.
But even for an adult, the standard practice when verifying a patient prior to doing almost anything is to ask them their name and their date of birth.  So... everything would stop if the date of birth didn't seem to match.

sjswerdloff reviewed (changes requested):
Change the PR name, at least.

Nowhere4Nothing committed:
Checks for UUID, localises Format date

Changed the dicon_utils to check for a UUID. Added private function that returns a true or false if UUID or not
Uses babel to format the date displayed to say ex: April 1 1990 do remove ambiguity with date
added a file that gitignore is not ignoring that needs to be ignored
